src/analysis/classic/warlock/demonology/modules/spells/Doomguard.tsx

```diff
@@ -0,0 +1,394 @@
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SpellLink from 'interface/SpellLink';
+import Events, {
+  CastEvent,
+  DamageEvent,
+  DeathEvent,
+  FightEndEvent,
+  Item,
+} from 'parser/core/Events';
+import SPELLS from 'common/SPELLS/classic/warlock';
+import { PerformanceMark } from 'interface/guide';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+import GEAR_SLOTS from 'game/GEAR_SLOTS';
+import TRINKETS from 'common/ITEMS/classic/trinkets';
+import ItemLink from 'interface/ItemLink';
+import enchanting from 'common/SPELLS/classic/enchanting';
+import { SpellInfo } from 'parser/core/EventFilter';
+import engineering from 'common/SPELLS/classic/engineering';
+import tailoring from 'common/SPELLS/classic/tailoring';
+import potions from 'common/SPELLS/classic/potions';
+import { formatDuration } from 'common/format';
+import Icon from 'interface/Icon';
+
+const relevantGearSlotsToCheckProcs: number[] = [
+  GEAR_SLOTS.TRINKET1,
+  GEAR_SLOTS.TRINKET2,
+  GEAR_SLOTS.HANDS,
+  GEAR_SLOTS.BACK,
+  GEAR_SLOTS.MAINHAND,
+];
+
+export type DoomguardData = {
+  inferno: {
+    summonEvent?: CastEvent;
+  };
+  doomguard: {
+    summonTimestamp: number;
+    deathTimestamp: number;
+    summonEvent?: CastEvent;
+    castEvents: CastEvent[];
+    totalDamage?: number;
+  };
+  summonedDemonSummary: JSX.Element;
+};
+
+export type SnapshotQualityEntry = {
+  item?: Item;
+  relatedBuffs: SpellInfo[];
+  issueWithItemOrEnchant: boolean;
+  snapshotQuality: QualitativePerformance;
+  snapshotSummary: JSX.Element;
+};
+
+export default class Doomguard extends Analyzer {
+  doomguardSummonData: DoomguardData = {
+    doomguard: {
+      summonTimestamp: 0,
+      deathTimestamp: 0,
+      castEvents: [],
+    },
+    inferno: {},
+    summonedDemonSummary: <></>,
+  };
+  snapshotQualityEntries: SnapshotQualityEntry[] = [];
+  snapshotAdvice: JSX.Element = (<></>);
+  private doomguardId: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.buildSnapshotEntries();
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SUMMON_DOOMGUARD),
+      this.onDoomguardSummonEvent,
+    );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell([SPELLS.SUMMON_INFERNO]),
+      this.onInfernoCast,
+    );
+    this.addEventListener(
+      Events.cast.by(this.doomguardId).spell([SPELLS.DOOM_BOLT]),
+      this.onDoomguardCast,
+    );
+    this.addEventListener(
+      Events.damage.by(this.doomguardId).spell([SPELLS.DOOM_BOLT]),
+      this.onDoomguardDamage,
+    );
+    this.addEventListener(Events.death, this.onDoomguardDeath);
+    this.addEventListener(Events.fightend, this.onFightEnd);
+  }
+
+  private buildSnapshotEntries() {
+    // Gear related buffs
+    relevantGearSlotsToCheckProcs.forEach((gearSlot) => {
+      const equippedItem: Item = this.selectedCombatant._combatantInfo.gear[gearSlot];
+      switch (gearSlot) {
```

Comment:
rather than using this loop/switch construction, you should split these out by slot in their own methods.

for example, the analysis for synapse springs could be:

```tsx
private synapseSpringsSnapshotEntry(): SnapshotQualityEntry {
  const equippedItem = this.selectedCombatant.hands;

  const synapseSpringsAvailable =
    equippedItem.onUseEnchant === engineering.SYNAPSE_SPRINGS_INTEL_BUFF.enchantId;

  return {
    item: equippedItem,
    snapshotQuality: QualitativePerformance.Fail, // NOTE: this is copied from your code but seems to be wrong since it is always a fail?
    relatedBuffs: [engineering.SYNAPSE_SPRINGS_INTEL_BUFF],
    issueWithItemOrEnchant: !synapseSpringsAvailable,
    snapshotSummary: synapseSpringsAvailable ? (
      <></>
    ) : (
      <li>
        <ItemLink id={equippedItem.id} details={equippedItem} icon={false}>
          <Icon icon={equippedItem.icon} />
        </ItemLink>{' '}
        is not enchanted with <SpellLink spell={engineering.SYNAPSE_SPRINGS} />. If you
        are not engineer, consider to change profession. This is by far the best one,
        especially as a warlock due to snapshot mechanics.
      </li>
    ),
  };
}